spec/models/manageiq/providers/vmware/cloud_manager_spec.rb

```diff
@@ -101,4 +101,20 @@
   it "#supported_catalog_types" do
     expect(@ems.supported_catalog_types).to eq(%w(vmware))
   end
+
+  describe 'snapshot operations' do
+    let(:vm) { FactoryGirl.create(:vm_vmware, :ext_management_system => @ems) }
+
+    context ".vm_create_snapshot" do
+      let(:snapshot_options) { {:name => 'name', :memory => false, :quiesce => false} }
+      let(:connection) { double("connection", :post_create_snapshot => "post_create_snapshot") }
+
+      it 'creates a snapshot' do
+        expect(@ems).to receive(:with_provider_connection).and_yield(connection)
+
+        expect(connection).to receive(:post_create_snapshot)
```

Comment:
When you add the `process_task` as suggested by @miha-plesko check the example [here](https://github.com/ManageIQ/manageiq-providers-vmware/blob/master/spec/models/manageiq/providers/vmware/cloud_manager/vm_spec.rb#L43) on how to mock the response.